adafruit_lis3dh/lis3dh.py

```diff
@@ -113,6 +115,33 @@ def acceleration(self):
 
         return (x / divider * 9.806, y / divider * 9.806, z / divider * 9.806)
 
+    def shake(self, shake_threshold=30, avg_count=10, total_delay=0.1):
+        """Detect when the accelerometer is shaken. shake_accel creates a list of
+        avg_count tuples, and adds each set of tuple members together, returning
+        a single tuple. avg takes that result and divides each member of that
+        tuple by avg_count to get the average of the data. Then total_accel
+        calculates the total acceleration by taking the square root of the sum of
+        the squares, i.e. sqrt(X*X + Y*Y + Z*Z). total_accel is compared to the
+        shake_threshold to determine if the device has been shaken.
+        Optional parameters:
+         shake_threshold - Increase or decrease to change shake sensitivity. This
+                           requires a minimum value of 10. 10 is the total
+                           acceleration if the board is not moving, therefore
+                           anything less than 10 will erroneously report a constant
+                           shake detected. (Default 30)
+         avg_count - The number of readings taken and used for the average
+                     acceleration. (Default 10)
+         total_delay - The total time in seconds it takes to obtain avg_count
+                       readings from acceleration. (Default 0.1)
+         """
+        shake_accel = (0, 0, 0)
+        for i in range(avg_count):
```

Comment:
Please put the internal explanation here after a #. The """ comment should only have the "public" info.